### ISSUE TITLE
Is it possible to bridge PPPoE IPv6

### ISSUE THREAD

**Issue by brshw** [cvmiller/v6brouter#1]
I get both IPv4 and IPv6 addresses from my ISP using PPPoE, and I use OpenWrt. Is it possible to set it up as a v6brouter?

IPv6 is set up automatically with the default network configuration with pppoe protocol:

```
network.wan.proto='pppoe'
```

**Comment by cvmiller** [comment]
Sorry, but I doubt it. I believe ebtables (the linux bridge that is used by v6brouter) will only bridge ethernet to ethernet, not ppp to ethernet.

But do you need it? If your upstream provider is providing an IPv6 address over PPPoE, are they not also providing Prefix Delegation (via DHCPv6-PD)?
Have you looked at the OpenWRT IPv6 networking page?
https://wiki.openwrt.org/doc/uci/network6

**Comment by brshw** [comment]
I do have globally routable IPv6 for my LAN. The reason I prefer this brouter set up is that it bypasses routing on weak hardware. Is there another way to avoid IPv6 routing for PPPoE?

**Comment by cvmiller** [comment]
It sounds like you are concerned about the performance of your router. Technically bridging should be faster than routing (since the packet remains unchanged when forwarding ethernet packets), but since both are happening in the kernel (with v6brouter/ebtables), I suspect you won't see much of a performance difference.

I just did some reading up on ebtables, and it _may_ supporting bridging between ppp and ethernet (which isn't just bridging since I had to make up a MAC address for the ethernet side, and remember it when it sees a packet come back in on the ethernet interface, and remove it, and put on a PPP header, recalc FCS, and such. So again, if you are concerned about performance, I suspect this isn't the solution for you.

That said, if you edit v6brouter script to point to your ppp-wan interface
OUTSIDE=ppp-wan

And run the script and see what happens. The easiest way you will know it is working is that your nodes on your LAN will have the same prefix as your current WAN interface.

HTH
